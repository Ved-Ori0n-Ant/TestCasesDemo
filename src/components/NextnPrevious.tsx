import React from 'react';
import {Button, StyleSheet, View} from 'react-native';
import {RootStackParamList} from '../navigation/MainStack';
import {StackActions, useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';

type props = {
  currentScreen: Stack;
  testID?: string;
};
const SCREENS = [
  'TextScreen',
  'TextTranslationScreen',
  'TextInputScreen',
  'ButtonScreen',
  'ImageScreen',
  'FlatListScreen',
];
export enum Stack {
    'TextScreen',
    'TextTranslationScreen',
    'TextInputScreen',
    'ButtonScreen',
    'ImageScreen',
    'FlatListScreen',
}

const NextnPrevious: React.FC<props> = ({currentScreen, testID}) => {
  const navigation =
    useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  const onPressNext = () => {
    navigation.dispatch(StackActions.replace(SCREENS[currentScreen + 1]));
  };
  const onPressPrevious = () => {
    navigation.dispatch(StackActions.replace(SCREENS[currentScreen - 1]));
  };
  return (
    <View style={styles.body} testID={testID || 'np_btn'}>
      <Button
        title={'Previous'}
        testID={'Previous_btn'}
        onPress={onPressPrevious}
        disabled={currentScreen <= 0}
      />
      <Button
        title={'Next'}
        testID={'Next_btn'}
        onPress={onPressNext}
        disabled={currentScreen >= SCREENS.length - 1}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: '#ddd',
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
});

export default NextnPrevious;
