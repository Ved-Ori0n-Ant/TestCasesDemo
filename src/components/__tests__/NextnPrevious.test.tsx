import React from 'react';
import NextnPrevious, {Stack} from '../NextnPrevious';
import {fireEvent, render} from '@testing-library/react-native';
import {NavigationContainer, StackActions} from '@react-navigation/native';

function renderWithNavigation(renderComponent: any) {
    return(
        render(
            <NavigationContainer>
                {renderComponent}
            </NavigationContainer>
        )
    )
}

const mockDispatch = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    dispatch: mockDispatch,
  }),
}));

describe('NextnPrevious Tests', () => {
  it('should render correctly', () => {
    const tree = renderWithNavigation(
      <NextnPrevious currentScreen={Stack.TextScreen} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have 2 Button', () => {
    const tree = renderWithNavigation(
      <NextnPrevious currentScreen={Stack.TextScreen} />,
    );
    const Next = tree.queryByTestId('Next_btn');
    const Previous = tree.queryByTestId('Previous_btn');

    expect(Next).not.toBeNull();
    expect(Previous).not.toBeNull();
  });

  it('should fireEvent', async () => {
    const tree = renderWithNavigation(
      <NextnPrevious currentScreen={Stack.TextInputScreen} />,
    );
    const Next = tree.getByTestId('Next_btn');
    const Previous = tree.getByTestId('Previous_btn');

    fireEvent.press(Next);
    expect(mockDispatch).toBeCalled();
    expect(mockDispatch).toBeCalledWith(StackActions.replace('ButtonScreen'));
    
    fireEvent.press(Previous);
    expect(mockDispatch).toBeCalled();
    expect(mockDispatch).toBeCalledWith(StackActions.replace('TextTranslationScreen'));
  });

  it('should disabled Previous', async () => {
    const tree = renderWithNavigation(
      <NextnPrevious currentScreen={Stack.TextScreen} />,
    );
    const Previous = tree.getByTestId('Previous_btn');
    const status = Previous.props.accessibilityState.disabled;
    expect(status).toBe(true);
  });
  it('should disabled Next', async () => {
    const tree = renderWithNavigation(
      <NextnPrevious currentScreen={Stack.FlatListScreen} />,
    );
    const Next = tree.getByTestId('Next_btn');
    const status = Next.props.accessibilityState.disabled;
    expect(status).toBe(true);
  });
});