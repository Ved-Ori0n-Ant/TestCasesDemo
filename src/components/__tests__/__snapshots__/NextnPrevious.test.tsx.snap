// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextnPrevious Tests should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ddd",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
  testID="np_btn"
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="Previous_btn"
  >
    <View
      style={
        [
          {},
          {},
        ]
      }
    >
      <Text
        disabled={true}
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Previous
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="Next_btn"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        disabled={false}
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
